### DISCUSSION TITLE
Separators between version number and release date

### DISCUSSION THREAD

**Discussion by yudai-nkt** [olivierlacan/keep-a-changelog#497] [Ideas]
This might be a possible duplicate of #59, but I have a question about what should be used between version number and release date.

There are several hyphen-shaped characters in Unicode, e.g.,

- &#x2010; (U+2010 HYPHEN)
- &ndash; (U+2013 EN DASH), or `&ndash;` for the sake of typing ease in HTML
- &mdash; (U+2014 EM DASH), or `&mdash;`

and each has its own role. An excerpt from [Oxford Dictionaries](https://en.oxforddictionaries.com/punctuation/hyphen) says

> Hyphens are used to link words and parts of words. They are not as common today as they used to be, but there are three main cases where you should use them:
> - in compound words
> - to join prefixes to other words
> - to show word breaks

and neither of these seems to apply for hyphen usage here. [Another excerpt](https://en.oxforddictionaries.com/grammar/dash-and-em-dash) says

> An em dash, or __long dash__, is used:
> - [...]
> - to show other kinds of break in a sentence where a comma, semicolon, or colon would be traditionally used:

and this is the case I think separators fall under. So should the hyphens be replaced by em dash (or possibly en dash)?

Since I'm not an expert of English, I can't say using hyphens is typographically incorrect. That being said, I would say hyphen looks weird IMHO and this would be worth discussing to make Keep a Changelog better and  more human-readable.

**Comment by yudai-nkt** [comment]
FWIW, this is how release heading changes:

> ## [1.0.0] - 2017-06-20

> ## [1.0.0]&mdash;2017-06-20

[1.0.0]: https://github.com/olivierlacan/keep-a-changelog/compare/v0.3.0...v1.0.0

**Comment by obosob** [comment]
I'd say the main reason it's common to use a simple ascii `-` over a Unicode char is to support text editors and terminals that don't support Unicode (though there are not many these days). Also the difference is much more subtle in a fixed-width font, so the value it adds is minimal (`-` vs `—`).

**Comment by brauliobz** [comment]
Also n-dash and m-dash are difficult to type. What are our options? `&mdash;`? `Alt + 0151` on Windows? `Ctrl+Shift+U+2014` on Linux? IMO this would be solved only if common Markdown began to accept `--` and `---` as n-dash and m-dash (see https://github.com/github/markup/issues/77 , for example).

**Comment by yudai-nkt** [comment]
Character entity references such as `&ndash;` or `&mdash;` are easy enough to type for me, but it doesn't seem to be the case for everyone...

>  Also the difference is much more subtle in a fixed-width font

Yes, but I think it's unlikely for Markdown files to be rendered in mono-spaced fonts unless we use such CSS stylesheets.

**Comment by axd1967** [comment]
An alternative is to inverse the fields and use ':'. This would give `YYYY-MM-DD : version`. See also https://github.com/olivierlacan/keep-a-changelog/issues/59#issuecomment-538630017
